File: 1/2.js
var fs = require('fs');
var lines = fs.readFileSync('1/in.txt', 'utf-8').split('\n').map(Number)

let cost = function(amt) {
  let needed = Math.floor(amt / 3) - 2
  if(needed <= 0) {
    return 0
  } else {
    return needed + cost(needed)
  }
}

console.log(lines.map(cost).reduce((a, b) => a + b))

File: 1/2.rb
def cost(amt)
  needed = (amt / 3) - 2
  if needed <= 0
    return 0
  else
    return needed + cost(needed)
  end
end

def run(lines)
  return (lines.map {|e| cost e }).reduce(0, :+)
end

lines = File.read('1/in.txt').split.map {|line| line.to_i }
puts run lines
